src/go/wsl-helper/cmd/kubeconfig.go

```diff
@@ -42,80 +46,174 @@ var kubeconfigCmd = &cobra.Command{
-			// We will only return nil in this path.
-			target, err := os.Readlink(linkPath)
-			if err != nil {
-				if errors.Is(err, os.ErrNotExist) {
-					fmt.Println("false")
-				} else if errors.Is(err, syscall.EINVAL) {
-					fmt.Printf("File %s exists and is not a symlink\n", linkPath)
-				} else {
-					fmt.Printf("%s\n", err)
-				}
-			} else if target == configPath {
-				fmt.Println("true")
-			} else {
-				// For a symlink pointing elsewhere, we assume we can overwrite.
-				fmt.Println("false")
-			}
-			return nil
+		linuxConfigDir := path.Join(homedir.HomeDir(), ".kube")
+		linuxConfig, err := readKubeConfig(filepath.Join(linuxConfigDir, "config"))
+		if err != nil && !errors.Is(err, os.ErrExist) {
+			return err
 		}
-		if enable {
-			err = os.Mkdir(configDir, 0o750)
-			if err != nil && !errors.Is(err, os.ErrExist) {
-				// The error already contains the full path, we can't do better.
-				return err
-			}
-			err = os.Symlink(configPath, linkPath)
-			if err != nil {
-				if errors.Is(err, os.ErrExist) {
-					// If it already exists, do nothing; even if it's not a symlink.
-					return nil
-				}
-				return err
-			}
-		} else {
-			// No need to create if we want to remove it
-			target, err := os.Readlink(linkPath)
-			if err != nil {
-				if errors.Is(err, os.ErrNotExist) {
-					return nil
-				}
-				return err
-			}
-			if target == configPath {
-				err = os.Remove(linkPath)
-				if err != nil && !errors.Is(err, os.ErrNotExist) {
-					return err
-				}
-			}
+
+		cleanConfig := removeExistingRDConfig(rdCluster, &linuxConfig)
+
+		kubeConfig, err := updateClusterIP(winConfig, *cleanConfig, rdNetworking)
+
+		var finalKubeConfigFile *os.File
+		if err := os.MkdirAll(linuxConfigDir, 0o750); err != nil {
+			return err
+		}
+		finalKubeConfigFile, err = os.Create(filepath.Join(linuxConfigDir, "config"))
+		if err != nil {
+			return err
+		}
+		defer finalKubeConfigFile.Close()
+		err = os.MkdirAll(linuxConfigDir, 0o750)
+		if err != nil && !errors.Is(err, os.ErrExist) {
+			// The error already contains the full path, we can't do better.
+			return err
+		}
+		err = yaml.NewEncoder(finalKubeConfigFile).Encode(kubeConfig)
+		if err != nil {
+			return err
 		}
 		return nil
 	},
 }
 
+func readKubeConfig(configPath string) (kubeConfig, error) {
+	var config kubeConfig
+	configFile, err := os.Open(configPath)
+	if err != nil {
+		return config, err
+	}
+	defer configFile.Close()
+	err = yaml.NewDecoder(configFile).Decode(&config)
+	if err != nil {
+		return config, err
+	}
+
+	return config, nil
+}
+
+// updateClusterIP reads the kube config from windows side it also
+// modifies the cluster's server host to an appropriate address.
+// It then merges the config with an exisiting configuration on
+// users distro and returns the merged config.
+func updateClusterIP(winConfig, linuxConfig kubeConfig, rdNetworking bool) (kubeConfig, error) {
+	ip, err := getClusterIP()
```

Comment:
We can now move `getClusterIP` into the `!rdNetworking` branch (because we don't need it otherwise, so we don't need to fail if it fails).